base/game_events/PsynergyStoneEvent.ts

```diff
@@ -0,0 +1,154 @@
+        const stone_sprite_x = this.data.hero.sprite.centerX;
+        const stone_sprite_y = this.data.hero.sprite.centerY - 24;
+
+        //Particle emitter
+        const emitter = this.game.add.emitter(stone_sprite_x, stone_sprite_y, 15);
+        const psynergy_particle_base = this.data.info.misc_sprite_base_list["psynergy_particle"];
+        const sprite_key = psynergy_particle_base.getSpriteKey("psynergy_particle");
+        emitter.makeParticles(sprite_key);
+        emitter.minParticleSpeed.setTo(-1, -1);
+        emitter.maxParticleSpeed.setTo(1, 1);
+        emitter.gravity = 100;
+        emitter.width = emitter.height = 20;
+        emitter.forEach((particle: Phaser.Sprite) => {
+            psynergy_particle_base.setAnimation(particle, "psynergy_particle");
+        });
+
+        emitter.start(false, 300, 250, 0);
+        const anim_key = psynergy_particle_base.getAnimationKey("psynergy_particle", "vanish");
+        emitter.forEach((particle: Phaser.Sprite) => {
+            particle.animations.play(anim_key, 10);
+            particle.animations.currentAnim.setFrame((Math.random() * particle.animations.frameTotal) | 0);
+        });
+
+        this.game.add
+            .tween(this.origin_npc.sprite.body)
+            .to({x: stone_sprite_x, y: stone_sprite_y}, 300, Phaser.Easing.Linear.None, true);
+        await this.set_text(GET_TEXT(hero_name));
+
+        //Recover PP and show status window
+        this.chars_status_window.show();
+        this.stone_pp_recovery();
+        this.chars_status_window.update_chars_info();
+        this.chars_status_window.update_position();
+        this.data.audio.play_se("battle/heal_1");
+        await this.set_text(RECOVERY_TEXT(hero_name));
+
+        emitter.destroy();
+        let closed: () => void;
+        this.chars_status_window.close(closed);
+
+        this.data.audio.play_se("misc/psynergy_stone_shatter");
+        const animation_key_shatter = this.origin_npc.sprite_info.getAnimationKey("stone", "shatter");
+        const animation_shatter = this.origin_npc.sprite.animations.getAnimation(animation_key_shatter);
+        animation_shatter.play().onComplete.addOnce(() => {
+            this.finish();
+        });
+    }
+
+    //Hide Stone sprite and return control to character
+    async finish() {
+        this.origin_npc.sprite.destroy();
+        await this.set_text(DISAPPEAR_TEXT);
+
+        this.origin_npc.visible = false;
+        this.data.hero.play(base_actions.IDLE);
+        this.running = false;
+        this.control_enable = false;
+        this.data.game_event_manager.force_idle_action = true;
+        --this.data.game_event_manager.events_running_count;
+        this.finish_events.forEach(event => event.fire(this.origin_npc));
+    }
+
+    stone_pp_recovery() {
+        this.data.info.party_data.members.forEach(member => {
+            member.current_pp = member.max_pp;
+        });
+    }
+
+    //Write text
+    async set_text(text) {
```

Comment:
In this case, it will be better if you use the `quick_next/kill_dialog` approach, in this way, you can have good control of the crystal that appears in the dialog and also create only one time the DialogManager instance.

Instantiate the `this.dialog_manager` object only once at the `_fire()` start.
If you want to show a dialog, call `this.dialog_manager.quick_next` function, see the options in order to control the crystal, and have a callback when the window is ready.
If you want to kill a running dialog, just call `this.dialog_manager.kill_dialog` (remember to set the `destroy_crystal` arg of `kill_dialog` args to true at least in the last call in order to delete all DialogManager crystal sprites).

See SummonEvent.ts if you want an example.